lib/backburner/worker.rb

```diff
@@ -36,20 +35,30 @@ def self.enqueue(job_class, args=[], opts={})
 
       begin
         response = nil
-        connection = Backburner::Connection.new(Backburner.configuration.beanstalk_url)
-        connection.retryable do
-          tube = connection.tubes[expand_tube_name(queue || job_class)]
-          serialized_data = Backburner.configuration.job_serializer_proc.call(data)
-          response = tube.put(serialized_data, :pri => pri, :delay => delay, :ttr => ttr)
+        connection_pool.with do |connection|
+          connection.retryable do
+            tube = connection.tubes[expand_tube_name(queue || job_class)]
+            serialized_data = Backburner.configuration.job_serializer_proc.call(data)
+            response = tube.put(serialized_data, :pri => pri, :delay => delay, :ttr => ttr)
+          end
         end
         return nil unless Backburner::Hooks.invoke_hook_events(job_class, :after_enqueue, *args)
-      ensure
-        connection.close if connection
       end
 
       response
     end
 
+    def self.connection_pool
+      pool = Thread.current[:beanstalkd_connection_pool]
+
+      unless pool
+        pool = ConnectionPool.new(Backburner.configuration.connection_pool_options) { Backburner::Connection.new(Backburner.configuration.beanstalk_url) }
+        Thread.current[:beanstalkd_connection_pool] = pool
```

Comment:
Is there a reason you are using a connection pool per thread? Would it not be better to share connections across threads? If you we using the threaded worker or a threaded web server you would see a lot of connections.